[victorpengmx] iP

victorpengmx opened the pull request:
# Albatross

> The problem with quotes found on the Internet is that they are often not true. - Galileo Gallilei

Albatross frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is
1. Download it from [here](https://github.com/victorpengmx/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

victorpengmx committed:
complete Level 0

victorpengmx committed:
complete level 1

victorpengmx committed:
complete level 2

victorpengmx committed:
complete level 3

victorpengmx committed:
Complete level 4

victorpengmx committed:
complete automated text ui testing

victorpengmx committed:
complete level 5

victorpengmx committed:
complete level 6 and fix issues with level 5

victorpengmx committed:
Complete Level-7

victorpengmx committed:
Merge branch 'branch-Level-7'

victorpengmx committed:
Complete level 8

victorpengmx committed:
Merge branch 'branch-level-8'

victorpengmx committed:
Merge branch 'branch-Level-8'

victorpengmx committed:
Merge branch 'master' of https://github.com/victorpengmx/ip

victorpengmx committed:
Add more OOP

victorpengmx committed:
Merge branch 'branch-A-MoreOOP'

victorpengmx committed:
Organize into packages

victorpengmx committed:
Merge branch 'branch-A-Packages'

victorpengmx committed:
Merge branch 'add-gradle-support'

victorpengmx committed:
Use Gradle

victorpengmx committed:
Merge branch 'branch-A-Gradle'

victorpengmx committed:
Add JUnit tests

victorpengmx committed:
Merge branch 'branch-A-JUnit'

victorpengmx committed:
Add and change some JavaDoc comments

victorpengmx committed:
Tweak code to NOT comply with coding standard since previous code was already compliant

victorpengmx committed:
Complete level 9

victorpengmx committed:
Merge branch 'branch-Level-9'

victorpengmx committed:
Merge branch 'branch-A-CodingStandard'

victorpengmx committed:
Merge branch 'branch-A-JavaDoc'

Vanessamae23 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,291 @@
+     * @throws DukeException if the command is invalid.
+     */
+    public static Command parse(String input) throws DukeException {
+        if (byeCommandChecker(input) != null) {
```
Perhaps can consider switch case

Vanessamae23 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+     *
+     * @param number The index of the task to be removed.
+     */
+    public void delete(int number) throws DukeException {
```
Is it better if we name it to deleteTask to be more descriptive

Vanessamae23 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,49 @@
+     * @param endDate End date of the event.
+     * @param done Whether the task is marked done or not.
+     */
+    public Event(String name, String startDate, String endDate, boolean done) {
```
is it better if we name it to `isDone` for the boolean flag

Vanessamae23 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    public String displayableForm() {
```
what is this supposed to do?

Vanessamae23 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Override
+    public String toString() {
```
maybe can consider `isDone` compared to `done` flag

Vanessamae23 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+        return this.index;
+    }
+
+    public Task getTask(int index) throws DukeException {
```
maybe can describe what this method does

Vanessamae23 reviewed (commented):
Looks good to me good job

shyanyong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+        }
+    }
+
+    public int getSize() {
```
Given that you've meticulously provided JavaDoc comments for all classes and methods elsewhere, is it intentional that there are missing JavaDoc comments in this particular method?

shyanyong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+    /**
+     * A task list that stores the user's tasks.
+     */
+    private ArrayList<Task> taskList;
```
Would naming the collection of task objects to a plural form such as 'tasks' be better?

shyanyong commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,44 @@
+        }
+    }
+
+    public boolean isExit() {
```
Same as getSize(), is the missing JavaDoc left out intentionally.

shyanyong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+
+    public static void main(String[] args) {
```
LGTM, I like how you formatted your code and split it into different classes.  Made it really clean and easy for reviewers to look through! 👍

victorpengmx committed:
Complete Level 10

victorpengmx committed:
Merge branch 'branch-Level-10'

victorpengmx committed:
Add assertions

Currently there are no assertions implemented.

Important assumptions in the code of some classes should be documented.

Let's add assertions for the Storage, ListCommand, and TaskList classes.

victorpengmx committed:
Improve Code Quality

Some methods are deeply nested
and some variable names make little sense.

This makes reading code more difficult.

Let's,
*change the names of these variables
*refactor these methods

victorpengmx committed:
Merge pull request #1 from victorpengmx/branch-A-Assertions

Add assertions

victorpengmx committed:
Merge branch 'branch-A-CodeQuality'

victorpengmx committed:
Merge pull request #2 from victorpengmx/branch-A-CodeQuality

Improve Code Quality

victorpengmx committed:
Merge branch 'master' of https://github.com/victorpengmx/ip

victorpengmx committed:
Add Help command

victorpengmx committed:
Merge branch 'branch-C-Help'

victorpengmx committed:
Improve Gui by ensuring text is not cut off, resize user and duke images

victorpengmx committed:
Merge branch 'branch-A-BetterGui'

victorpengmx committed:
Update user guide and fix bugs

victorpengmx committed:
Merge branch 'branch-A-UserGuide'

victorpengmx committed:
Update User Guide
